application/src/js/all/_secondary_sources.js

```diff
@@ -134,36 +134,27 @@ SecondarySource.prototype.showAddSourceButton = function(show) {
 
 SecondarySource.prototype.addSourceButtonClicked = function (event) {
   // `remove_data_source` hooks into the application.cms.forms.DataSourceForm field.
-  this.fieldset.querySelector('.js-remove-data-source').value = "false"
+  this.fieldset.querySelector('.js-remove-data-source').querySelector('input').checked = false;
 
   this.setHidden(false)
   event.preventDefault()
 };
 
-SecondarySource.prototype.removeSourceButtonClicked = function(event) {
-
-  var fields = this.fieldset.querySelectorAll('input, textarea, option')
-
-  for (var i = 0; i < fields.length; i++) {
-    if (fields[i].type !== "hidden") { fields[i].value = "" }
-    fields[i].removeAttribute('checked')
-    fields[i].removeAttribute('selected')
-  };
-
+SecondarySource.prototype.removeSourceButtonClicked = function (event) {
   this.setHidden(true)
 
   // `remove_data_source` hooks into the application.cms.forms.DataSourceForm field.
-  this.fieldset.querySelector('.js-remove-data-source').value = "true"
+  this.fieldset.querySelector('.js-remove-data-source').querySelector('input').checked = true;
```

Comment:
We can combine these into `this.fieldset.querySelector('.js-remove-data-source input').checked = true;`